src/status_im2/contexts/syncing/enter_sync_code/view.cljs

```diff
@@ -0,0 +1,84 @@
+(ns status-im2.contexts.syncing.enter-sync-code.view
+  (:require [clojure.string :as string]
+            [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [reagent.core :as reagent]
+            [react-native.core :as rn]
+            [react-native.clipboard :as clipboard]
+            [utils.i18n :as i18n]
+            [status-im2.contexts.syncing.enter-sync-code.style :as style]
+            [quo2.components.icon :as icon]
+            [utils.debounce :as debounce]
+            [utils.re-frame :as rf]
+            [status-im2.contexts.syncing.utils :as sync-utils]))
+
+(defn view
+  []
+  (let [default-value           (reagent/atom nil)
+        clipboard               (reagent/atom nil)
+        submit-button-disabled? (reagent/atom true)
+        profile-color           (rf/sub [:profile/customization-color])]
+    (fn []
+      (clipboard/get-string #(reset! clipboard %))
+      (let [invalid?           false
+            show-paste-button? (string/blank? @default-value)]
+        [:<>
+         [rn/view
+          {:style style/label-container}
+          [quo/text
+           {:style  style/label-pairing
+            :weight :medium
+            :size   :paragraph-2}
+           (i18n/label :t/type-pairing-code)]]
+         [rn/view {:style style/container-text-input}
+          [rn/view {:style (style/text-input-container invalid?)}
+           [rn/text-input
```

Comment:
we should be using a `quo` input component here no?

Reply:
<img width="320" alt="Screenshot 2023-08-02 at 10 04 32" src="https://github.com/status-im/status-mobile/assets/22799766/42991899-f8aa-4fe2-b825-3107374bf22d">

Reply:
`quo` input does not support resizing the input container when text grows... hence had to do a custom approach here.
I can make a follow up issue to modify `quo` input to implement this behaviour over there.

Reply:
yeah, I see that in the description now. I understand this issue is important for QA to use so this is not blocking for their sake.  However, we should avoid working in this format in the future where possible and prioritise getting the sub components created first as it's the better long term strategy (and the strategy the team has agreed upon).

Please create a follow up issue to 1. create the sub component properly and 2. use this component here instead.

Reply:
Yep, also Add contacts view also does a similar thing, I'll make sure I capture that as well in the new issue.

Reply:
Oh good spot @siddarthkay! Thanks! 🙏

Reply:
One thing to note here is that the input component in figma is still not updated with this behaviour in their design system cc @pedro-et @mariocnovoa 
ref : https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=57-5646&mode=design&t=aWe8hcvX8WG7ZOUG-0
The behaviour in question is that the `input` component expands to incorporate text as seen in enter scan sync code Design.

Reply:
issue : https://github.com/status-im/status-mobile/issues/16858